defmt/src/impls/core_/panic.rs

```diff
@@ -0,0 +1,28 @@
+use core::panic;
+
+use super::*;
+
+impl<'a> Format for panic::PanicInfo<'a> {
+    fn format(&self, f: Formatter) {
+        crate::write!(f, "panicked at {}:", self.location());
+        // TODO: consider supporting self.message() once stabilized, or add a crate feature for
+        // conditional support
+
+        #[allow(deprecated)] // Deprecated on nightly
+        if let Some(msg) = self.payload().downcast_ref::<&'static str>() {
+            crate::write!(f, "\n{=str}", msg);
+        }
```

Comment:
Should we write something else in case the payload is _not_ a static string? Something like `<panic info>` maybe?